Thanks @jonringer. I'm getting test failures. Does Nix(OS) sandboxing actually hide interfaces from the sandbox entirely? It could explain this.

```
=================================== FAILURES ===================================
_____________________________ test_cli_main_basic ______________________________
tests/test_cli.py:20: in test_cli_main_basic
    assert run_cmd(BASE_CMD) == get_mac_address()
E   AssertionError: assert '' == None
E    +  where '' = run_cmd(['/nix/store/s7vw8y02cqzx8bnjxl4bkhlnwvz6ws1s-python3-3.7.6/bin/python3.7', '-m', 'getmac'])
E    +  and   None = get_mac_address()
____________________________ test_cli_main_verbose _____________________________
tests/test_cli.py:24: in test_cli_main_verbose
    assert get_mac_address() in run_cmd(BASE_CMD + ["--verbose"])
E   TypeError: 'in <string>' requires string as left operand, not NoneType
_____________________________ test_cli_main_debug ______________________________
tests/test_cli.py:28: in test_cli_main_debug
    assert get_mac_address() in run_cmd(BASE_CMD + ["--verbose", "--debug"])
E   TypeError: 'in <string>' requires string as left operand, not NoneType
________________________ test_cli_multiple_debug_levels ________________________
tests/test_cli.py:49: in test_cli_multiple_debug_levels
    assert get_mac_address() in run_cmd(BASE_CMD + ["-v", "-dd"])
E   TypeError: 'in <string>' requires string as left operand, not NoneType

----------------------
```

I'll keep taking a look (maybe we can selectively disable just these tests), but of course any tips are appreciated. Sincere thanks for all the effort you took to review all of this multiple times. I really appreciate the pointers.

--- a/pkgs/development/python-modules/getmac/default.nix
+++ b/pkgs/development/python-modules/getmac/default.nix
@@ -1,0 +1,26 @@
+{ lib, buildPythonPackage, fetchFromGitHub
+, pytest, pytest-benchmark, pytest-mock }:
+
+buildPythonPackage rec {
+  pname = "getmac";
+  version = "0.8.2";
+
+  src = fetchFromGitHub {
+    owner = "GhostofGoes";
+    repo = "getmac";
+    rev = version;
+    sha256 = "08d4iv5bjl1s4i9qhzf3pzjgj1rgbwi0x26qypf3ycgdj0a6gvh2";
+  };
+
+  checkInputs = [ pytest pytest-benchmark pytest-mock ];
+  checkPhase = ''
+    pytest --ignore tests/test_cli.py
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/GhostofGoes/getmac";
+    description = "Pure-Python package to get the MAC address of network interfaces and hosts on the local network.";
+    license = licenses.mit;
+    maintainers = with maintainers; [ colemickens ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -690,6 +690,8 @@ in {
 
   geoip2 = callPackage ../development/python-modules/geoip2 { };
 
+  getmac = callPackage ../development/python-modules/getmac { };
+
   gidgethub = callPackage ../development/python-modules/gidgethub { };
 
   gin-config = callPackage ../development/python-modules/gin-config { };

